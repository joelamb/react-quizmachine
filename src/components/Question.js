import React from 'react';
import { decode } from 'he';

class Question extends React.Component {
  constructor(props) {
    super(props);
  }

  render() {
    const { question, answers, difficulty, handleClick, fetchQuestion } = this.props;

    return (
      <article>
        {question.question &&
          <React.Fragment>
            <p>{decode(question.question)}</p>
            <ul>
              {answers.map(answer => {
                return <li key={answer} > <button onClick={() => { handleClick(answer, difficulty, question.correct_answer); fetchQuestion(); }}>{decode(answer)}</button></li>
              })}
            </ul>
          </React.Fragment>
        }
      </article >

    )
  }
}

export default Question;