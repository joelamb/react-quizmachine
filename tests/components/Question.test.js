import React from 'react';
import Question from '../../src/components/Question';
import renderer from 'react-test-renderer';

describe('Question', () => {
  test('renders correctly', () => {
    const tree = renderer
      .create(<Question question={{
        category: "Sports",
        correct_answer: "Duck",
        difficulty: "medium",
        incorrect_answers: ["Bye", "Beamer", "Carry"],
        question: "What cricketing term denotes a batsman being dismissed with a score of zero?",
        type: "multiple"
      }} answers={["Bye", "Beamer", "Duck", "Carry"]} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  })
})