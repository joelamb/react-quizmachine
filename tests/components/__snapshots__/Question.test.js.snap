// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Question renders correctly 1`] = `
<article
  className="question-holder"
>
  <p
    className="question"
  >
    What cricketing term denotes a batsman being dismissed with a score of zero?
  </p>
  <ul
    className="answers"
  >
    <li
      className="answer btn"
      onClick={[Function]}
    >
      Bye
    </li>
    <li
      className="answer btn"
      onClick={[Function]}
    >
      Beamer
    </li>
    <li
      className="answer btn"
      onClick={[Function]}
    >
      Duck
    </li>
    <li
      className="answer btn"
      onClick={[Function]}
    >
      Carry
    </li>
  </ul>
</article>
`;
